### PR TITLE
[azeventhubs] Fixing issue with checkpoint/ownership blob casing that would affect compatibility

### LINKED ISSUE #21352: [Event Hubs] Checkpoint store should use lowercase names in blob paths
# Summary

The contract for the prefix used for naming blobs associated with ownership and checkpoint information is that all elements should be treated as lower-case.  Currently, the Go checkpoint store implementation is not adjusting case for the namespace, Event Hub, nor consumer group.  This should be corrected to allow for cross-language use of checkpoint data.

# Success criteria

- The path prefix used for blobs is lower-case and culture neutral.  
- If necessary, add fallback logic to allow use of the existing format and avoid a breaking change.

**Example:**

Given the following data:
- Fully Qualified Namespace:  `myNamespace.servicebus.windows.net`
- Event Hub: `myEventHub`
- Consumer Group: `$Default`

The name of the ownership blob for partition 0 should be:  `mynamespace.servicebus.windows.net/myeventhub/$default/ownership/0`

# References and Resources

- [blob_store.go::prefixForCheckpointBlobs](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azeventhubs/checkpoints/blob_store.go#L277)
- [blob_store.go::prefixForOwnershipBlobs](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azeventhubs/checkpoints/blob_store.go#L295C16-L295C16)

### PR DESCRIPTION
**Pull request by richardpark-msft**
- Fixing casing issue with blobs by adding an in-place migration to the proper lowercase blob path.
- Also adding in tests of all sorts - live and stress.

Fixes #21352

### CODE REVIEW

File: sdk/messaging/azeventhubs/checkpoints/blob_store.go [hunk 1]
```diff
@@ -94,48 +109,80 @@ func (b *BlobStore) ListCheckpoints(ctx context.Context, fullyQualifiedNamespace
...
+	if correctPrefix != legacyPrefix {
+		// we've got a casing difference - we need to make sure we look at both possible
+		// casings because of our previous bug where we didn't ToLower() the blob name by
+		// default.
+		if err := addCheckpointsToMap(legacyPrefix, true); err != nil {
+			return nil, err
 		}
 	}
 
+	if err := addCheckpointsToMap(correctPrefix, false); err != nil {
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/checkpoints/blob_store.go:168]: You could condense this into `addCheckpointsToMap(correctPrefix, correctPrefix != legacyPrefix)` and remove the previous block.

File: sdk/messaging/azeventhubs/checkpoints/blob_store.go [hunk 2]
```diff
@@ -146,54 +193,102 @@ func (b *BlobStore) ListOwnership(ctx context.Context, fullyQualifiedNamespace s
...
+	if legacyPrefix != correctPrefix {
+		// we've got a casing difference - we need to make sure we look at both possible
+		// casings because of our previous bug where we didn't ToLower() the blob name by
+		// default.
+		if err := addOwnershipsToMap(legacyPrefix, true); err != nil {
+			return nil, err
 		}
 	}
 
+	if err := addOwnershipsToMap(correctPrefix, false); err != nil {
```
Comment (jhendrixMSFT) [inline sdk/messaging/azeventhubs/checkpoints/blob_store.go:257]: Same thing as above.

File: sdk/messaging/azeventhubs/checkpoints/blob_store.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/messaging/azeventhubs/checkpoints/blob_store.go:168]: Ugh maybe not.

**Review by jhendrixMSFT — APPROVED** [review]

**Review by RickWinter — APPROVED** [review]

**Pull request merged**